soroban-env-host/src/test/stellar_asset_contract.rs

```diff
@@ -666,14 +690,154 @@ fn test_transfer_with_issuer() {
 
     // Issuer transfers to self
     contract
-        .transfer(&issuer, issuer.address(&test.host), 100)
+        .transfer(&issuer, issuer.address(&test.host), i64::MAX.into())
+        .unwrap();
+    assert_eq!(
+        test.host.get_events().unwrap().0,
+        vec![contract.test_event(
+            test_vec![
+                &test.host,
+                Symbol::try_from_small_str("transfer").unwrap().to_val(),
+                issuer.address(&test.host),
+                issuer.address(&test.host),
+                &token_name,
+            ],
+            (i64::MAX as i128).try_into_val(&test.host).unwrap()
+        )]
+    );
+}
+
+#[test]
+fn test_cap_67_transfer_with_muxed_accounts() {
+    let test = StellarAssetContractTest::setup(function_name!());
+    // Enable invocation metering to get the events to reset automatically on
+    // every contract call.
+    test.host.enable_invocation_metering();
+    let admin = TestSigner::account(&test.issuer_key);
+    let contract = test.default_stellar_asset_contract();
+
+    let user = TestSigner::account(&test.user_key);
+    let user_2 = TestSigner::account(&test.user_key_2);
+    test.create_default_account(&user);
+    test.create_default_account(&user_2);
+    test.create_default_trustline(&user);
+    test.create_default_trustline(&user_2);
+
+    contract
+        .mint(&admin, user.address(&test.host), 100_000_000)
         .unwrap();
 
-    let events = test.host.get_events().unwrap().0;
-    check_event_name(events[0].clone(), &contract.address, "mint");
-    check_event_name(events[1].clone(), &contract.address, "transfer");
-    check_event_name(events[2].clone(), &contract.address, "burn");
-    check_event_name(events[3].clone(), &contract.address, "transfer");
+    let transfer_symbol = Symbol::try_from_small_str("transfer").unwrap().to_val();
+    let token_name = contract.name().unwrap();
+
+    // Transfer some balance from user 1 to user 2, both source and destination
+    // are muxed.
+    contract
+        .transfer_muxed(
```

Comment:
You already have tests that make sure you can't construct an Address object from a liquidity pool or claimable balance id right?

Reply:
Yeah, these are in `address.rs` tests, and I'm also adding tests for covering this e2e in core as well.